moveit_ros/perception/handeye_calibration_target/src/handeye_aruco_target.cpp

```diff
@@ -0,0 +1,301 @@
+{
+  std::vector<std::string> dictionary_ids;
+  for (const std::pair<const std::string, cv::aruco::PREDEFINED_DICTIONARY_NAME>& name : marker_dictionaries_)
+    dictionary_ids.push_back(name.first);
+  return dictionary_ids;
+}
+
+bool HandEyeArucoTarget::setTargetIntrinsicParams(const int& markers_x, const int& markers_y, const int& marker_size,
+                                                  const int& separation, const int& border_bits,
+                                                  const std::string& dictionary_id)
+{
+  if (markers_x != markers_x_ || markers_y != markers_y_ || marker_size != marker_size_ || separation != separation_ ||
+      border_bits != border_bits_ || marker_dictionaries_.find(dictionary_id)->second != dictionary_id_)
+    ROS_DEBUG_STREAM_NAMED(LOGNAME, "Set target intrinsic params: "
+                                        << "\n"
+                                        << "markers_x_ " << std::to_string(markers_x) << "\n"
+                                        << "markers_y_ " << std::to_string(markers_y) << "\n"
+                                        << "marker_size " << std::to_string(marker_size) << "\n"
+                                        << "separation " << std::to_string(separation) << "\n"
+                                        << "border_bits" << std::to_string(border_bits) << "\n"
+                                        << "dictionary_id " << dictionary_id << "\n");
+  if (markers_x > 0 && markers_y > 0 && marker_size > 0 && separation > 0 && border_bits > 0 && !dictionary_id.empty())
+  {
+    std::lock_guard<std::mutex> lck(aruco_mtx_);
+    markers_x_ = markers_x;
+    markers_y_ = markers_y;
+    marker_size_ = marker_size;
+    separation_ = separation;
+    border_bits_ = border_bits;
+
+    const auto& it = marker_dictionaries_.find(dictionary_id);
+    if (it != marker_dictionaries_.end())
+      dictionary_id_ = it->second;
+    else
+    {
+      ROS_ERROR_STREAM_NAMED(LOGNAME, "Invalid marker dictionary name.");
+      return false;
+    }
+  }
+  else
+  {
+    ROS_ERROR_STREAM_NAMED(LOGNAME, "Invalid target intrinsic params.");
+    return false;
+  }
+
+  return true;
+}
+
+bool HandEyeArucoTarget::setTargetDimension(const double& marker_measured_size,
+                                            const double& marker_measured_separation)
+{
+  if (marker_measured_size != marker_size_real_ || marker_measured_separation != marker_separation_real_)
+    ROS_DEBUG_STREAM_NAMED(LOGNAME, "Set target real dimensions: "
+                                        << "\n"
+                                        << "marker_measured_size " << std::to_string(marker_measured_size) << "\n"
+                                        << "marker_measured_separation " << std::to_string(marker_measured_separation)
+                                        << "\n");
+
+  if (marker_measured_size > 0 && marker_measured_separation > 0)
+  {
+    std::lock_guard<std::mutex> lck(aruco_mtx_);
```

Comment:
avoid abbreviations per moveit style for variable names
```suggestion
    std::lock_guard<std::mutex> aruco_lock(aruco_mtx_);
```

Reply:
Agreed. Changed them to more readable names.